etl/steps/data/grapher/un/2023-01-24/un_sdg.py

```diff
@@ -0,0 +1,252 @@
+        publication_date=ds_garden.metadata.sources[0].publication_date,
+        publication_year=ds_garden.metadata.sources[0].publication_year,
+        published_by=ds_garden.metadata.sources[0].published_by,
+        publisher_source=df_gr["source"].iloc[0],
+    )
+
+    df_gr["meta"] = VariableMeta(
+        title=df_gr["variable_name_meta"].iloc[0],
+        description=source_desc_out,
+        sources=[source],
+        unit=df_gr["long_unit"].iloc[0].lower(),
+        short_unit=df_gr["short_unit"].iloc[0],
+        additional_info=None,
+    )
+
+    df_gr["variable"] = underscore(df_gr["variable_name"].iloc[0][0:254])
+
+    df_gr = df_gr[["country", "year", "value", "variable", "meta"]].copy()
+    # convert integer values to int but round float to 2 decimal places, string remain as string
+    df_gr["value"] = df_gr["value"].apply(value_convert)
+    df_gr = df_gr.set_index(["year", "country"])
+
+    return df_gr
+
+
+def create_dataframe_with_variable_name(dataset: Dataset, tab: str) -> pd.DataFrame:
+
+    cols_keep = [
+        "country",
+        "year",
+        "seriescode",
+        "seriesdescription",
+        "variable_name",
+        "variable_name_meta",
+        "value",
+        "source",
+        "long_unit",
+        "short_unit",
+    ]
+
+    tab_df = pd.DataFrame(dataset[tab]).reset_index()
+    cols_meta = ["indicator", "seriesdescription", "seriescode"]
+    cols = ["indicator", "seriescode"]
+    if tab_df.shape[1] > 11:
+        col_list = sorted(tab_df.columns.to_list())
+        drop_cols = [
+            "country",
+            "year",
+            "goal",
+            "target",
+            "indicator",
+            "seriescode",
+            "seriesdescription",
+            "value",
+            "source",
+            "long_unit",
+            "short_unit",
+            "variable_name",
+        ]
+        dim_cols = [x for x in col_list if x not in drop_cols]
+        cols_meta_dim = cols_meta + dim_cols
+        cols_dim = cols + dim_cols
+        tab_df["variable_name_meta"] = tab_df[cols_meta_dim].agg(" - ".join, axis=1)
+        tab_df["variable_name"] = tab_df[cols_dim].agg(" - ".join, axis=1)
+        tab_df = tab_df[cols_keep]
+        tab_df["seriescode"] = tab_df["seriescode"].str.lower()
+    else:
+        tab_df["variable_name_meta"] = tab_df[cols_meta].agg(" - ".join, axis=1)
+        tab_df["variable_name"] = tab_df[cols].agg(" - ".join, axis=1)
+        tab_df = tab_df[cols_keep]
+        tab_df["seriescode"] = tab_df["seriescode"].str.lower()
+
+    return tab_df
+
+
+def load_clean_source_mapping() -> Dict[str, str]:
+    """
+    Load the existing json which maps the raw sources to a cleaner version of the sources.
+    """
+    with open(paths.directory / "un_sdg.sources.json", "r") as f:
+        sources = json.load(f)
+        return cast(Dict[str, str], sources)
+
+
+@cache
+def get_metadata_link(indicator: str) -> str:
```

Comment:
Just a warning here - steps ideally shouldn't use data from other sources than snapshots. If those URLs suddenly stop working, ETL would start failing (or if they change it, we might end up with different metadata).

We've been using it with old SDGs and there weren't any problems, so no need to change it now, just be aware of it.

Reply:
That's a very good point, I don't think they are likely to change it, but you're right it's a bit dangerous. I will make a note to grab the metadata and store it in a snapshot for the next update.